jina/optimizers/flow_runner.py

```diff
@@ -24,13 +24,17 @@ def __init__(
         :param documents: iterator with list or generator for getting the documents
         :param batch_size: batch size used in the flow
         :param task: task of the flow which can be `index` or `search`
```

Comment:
task is better passed as argument of `run` I think, like this the `FlowRunner` instance is just dependant of the `Flow`. `task` is just syntax sugar for querying in jina

Reply:
I am not sure of a situation where I will use the same flow for both `index` and `query`. If a flow is always linked to a task, let's go with the current design.

Reply:
a Flow is by design not linked to a task

Reply:
but it is okey leave it like this